### PR TITLE
feat: Initial implementation

### PR DESCRIPTION
**Pull request by mykola-mokhnach**

### CODE REVIEW

File: .github/workflows/publish.js.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: lts/*
+    - run: |
+        npm install -g appium@next
```
Comment (KazuCocoa) [inline .github/workflows/publish.js.yml:18]:
```suggestion
        npm install -g appium
```
? :P

File: lib/mixins/proxy.js [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+  } catch (e) {
+    throw new Error(
+      `Cannot find any free port to forward candidate Devtools sockets ` +
+      `in range ${startPort}..${endPort}`
+    );
+  }
+  for (const socketName of socketNames) {
+    const remotePort = socketName.replace(/^@/, '');
+    try {
+      await adb.adbExec(['forward', `tcp:${localPort}`, `localabstract:${remotePort}`]);
```
Comment (KazuCocoa) [inline lib/mixins/proxy.js:150]: the same internally, but `adb.forwardAbstractPort`?

**Review by KazuCocoa — APPROVED** [review]
left minor comments, otherwise looks good overall to me.

btw, shall we move this in appium repo? (am ok to maintain this, btw. not sure if it should be in `appium/appium` tho)

File: .github/workflows/publish.js.yml (hunk 1 above)
Comment (mykola-mokhnach) [reply .github/workflows/publish.js.yml]: fixed

File: lib/mixins/proxy.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/mixins/proxy.js]: changed

**Comment by mykola-mokhnach** [comment]
> left minor comments, otherwise looks good overall to me.
> 
> btw, shall we move this in appium repo? (am ok to maintain this, btw. not sure if it should be in `appium/appium` tho)

thanks for reviewing it. Lets keep the plugin here for now. It still needs an extensive testing and validation. Also, I need to write a proper readme with usage examples.

**Pull request merged**